addons/account/models/mail_scheduled_message.py

```diff
@@ -0,0 +1,176 @@
+                'pdf_report': self.env['ir.actions.report'].browse(payload['pdf_report']),
+                'mail_template': self.env['mail.template'].browse(payload['mail_template']),
+            }
+            if payload['sending_methods'] and 'email' in payload['sending_methods']:
+                payload_attachments_widget = payload.get('mail_attachments_widget', [])
+                attachments_name_from_payload = [d.get('name') for d in payload_attachments_widget]
+                manual_attachments_widget_data = [
+                    {
+                        'id': attachment.id,
+                        'name': attachment.name,
+                        'mimetype': attachment.mimetype,
+                        'placeholder': False,
+                        'manual': True,
+                    }
+                    for attachment in scheduled_message.attachment_ids
+                    if attachment.name not in attachments_name_from_payload
+                ]
+                sending_settings = {
+                    **{k: v for k, v in sending_settings.items() if k in sending_settings_whitelist},
+                    'mail_partner_ids': scheduled_message.partner_ids.ids,
+                    'mail_subject': scheduled_message.subject,
+                    'mail_body': scheduled_message.body,
+                    'mail_attachments_widget': payload_attachments_widget + manual_attachments_widget_data,
+                }
+
+            # The generate and send invoice logic (including any EDI sending methods) with extra manual attachment if any
+            attachments = self.env['account.move.send']._generate_and_send_invoices(
+                move_id,
+                payload.get('allow_fallback_pdf'),
+                **{k: v for k, v in sending_settings.items() if k in sending_settings_whitelist},
+            )
+            if attachments and notification_parameters.get('from_move_send_wizard') and scheduled_message.is_note:
+                body = scheduled_message.body.replace('scheduled to send', _('sent'))
+                move_id.message_post(
+                    subject=scheduled_message.subject,
+                    body=body,
+                    subtype_xmlid='mail.mt_note',
+                )
+            scheduled_message.unlink()
+
+        if remained_scheduled_messages:
+            super(MailScheduledMessage, remained_scheduled_messages)._post_message(raise_exception)
+
+    # ------------------------------------------------------
+    # Business Methods
+    # ------------------------------------------------------
+
+    @api.model
+    def _sending_settings_whitelist(self):
+        """ Sending Settings that can be used when generating and sending invoices.
+        """
+        return {
+            'sending_methods',
+            'invoice_edi_format',
+            'extra_edis',
+            'pdf_report',
+            'author_user_id',
```

Comment:
When in doubt, either order alphabetically, either make section separated with comments telling what are the usage of various keys :)